### PR TITLE
feature/ATC-18

### PR DESCRIPTION
**Pull request by n8rzz**
This PR supersedes #19 and #35.  #35 will be closed in favor of this PR and the features for #19 will be included with this branch since they are so closely related.

### CODE REVIEW

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js:281]: @erikquinn This line has me totally stumped.  By including this line it actually causes aircraft to spawn closer to the controlled airspace?  Comparing this version to what I have in `gh-pages`, the spawn points look to be the same.  I'm not too sure on what was going on here originally, though, so I could use another set of eyes on this if you have some time.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js:311]: @erikquinn I see this number _a lot_ throughout the app.  What is it for?  Is it 3.6 seconds (my guess) or something else?  I would like to add this to a not-yet-created constants file and make this a little bit easier to understand when reading through.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [inline src/assets/scripts/airport/Route/RouteModel.js:24]: @erikquinn This class will likely need to be extended to work with the fms, once I get to that part.  Could you explain the route format a little but, or maybe correct me where I'm wrong.

A routes takes the shape `ORIGIN_NAME.PROCEDURE_NAME.DESTINATION_NAME`, but I've seen in to comments (somewhere) references to routes in the shape of `ORIGIN_NAME.PROCEDURE_NAME.DESTINATION_NAME..ORIGIN_NAME.PROCEDURE_NAME.DESTINATION_NAME`.

What is the second shape used for?  Is it actually being used within the app somewhere?  Is this just an extended route, like a star to an airway? or an airway to a star?

**Comment by n8rzz** [comment]
@erikquinn I've left some questions for you.  This is still a WIP but I feel like the `AirportModel` and `ArrivalBase` files are starting to look a lot cleaner.  There is a crapload of new stuff in there as well that now makes the concerns a little clearer.  Some things of note so far:
- `AirportModel` offloads maintenance of `StandardRoutes` to the `StandardRouteCollection`, which provides methods to get at specific information about a route.
- `StandardWaypointModel` now contains position information obtained from the `FixCollection`. This is a static class that I'm still not that happy with.  It works and it's tested, it just feels duplicative.  But, I suppose, they are separate concerns.  I don't know, this strategy will probably change in the future.
- When a `StandardRoute` route is added to the collection, we also calculate some position and distance information that makes spawn calculations _a lot_ easier. (That can be seen at work in the `ArrivalBase`.
- I've started pulling out magic numbers:  

``` javascript
// old
const altitude = 1000;

// new 
const DEFAULT_SPAWN_ALTITUDE_MIN = 10000;
const altitude = DEFAULT_SPAWN_ALTITUDE_MIN;
```

This does a few things; first it makes the `const altitude` line infinitely more readable. second, if we ever decide to change that default altitude, you can do so _in one place_.  I do need your help in figuring out what some of the magic numbers actually mean, and I've left comments for a lot of them throughout the app.  There are a few specifically here that I tagged you on already.

Those are some of the big items here.  There is still a lot to do with this branch, but it's getting closer and starting to feel a lot cleaner.  I'll continue working on the Arrival files and will hopefully be on to the Departure stuff by the end of the weekend.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:281]:
``` javascript
    /** Backfill STAR routes with arrivals closer than the spawn point
     ** Aircraft spawn at the first point defined in the route of the entry in
     ** "arrivals" in the airport json file. When that spawn point is very far
     ** from the airspace boundary, it obviously takes quite a while for them
     ** to reach the airspace. This function spawns (all at once) arrivals along
     ** the route, between the spawn point and the airspace boundary, in order to
     ** ensure the player is not kept waiting for their first arrival aircraft.
     */
    preSpawn: function() {
```

For each stream of arrivals, the process was:
1. Count the distance `d` along the route between the spawn point and airspace boundary
2. Determine the MIT (miles-in-trail) at which to spawn aircraft (`speed/frequency`)
3. Count `n` number of miles along the route from the spawn point toward the airspace boundary, and spawn an aircraft.
4. Repeat step 3 until `n > d`.

---

So really, I'm not sure quite what you're asking, but here's what I see:
- old lines 145-146 are related to step 3 (more specifically, storing the positions at which to spawn aircraft)
- new lines 301-302 are related to step 1, and add together:
  - the distance along the route from the spawn point to the last fix still outside of the airspace
  - the distance between the last fix still outside the airspace and the airspace boundary

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:311]: The `frequency` property is specified in acph (aircraft per hour), since that is a common (and obviously the most important) metric used in macro-level traffic management... but the game's time is all dealt with in seconds. This is just a way to translate `aircraft / hour` into `seconds / aircraft`.

Perhaps you're now realizing the magic number comes from:
`1sec * (60sec/min) * (60min/hr) ==> 3600sec/hr` 😜

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [reply src/assets/scripts/airport/Route/RouteModel.js:24]: It's for when there ISN'T any procedure to link two points. For example:
`Airway J123: ["A", "B", "C", "D", "E"]`
You could have `A.J123.E` or simply `A..E` to cut the corner, or `A..X` for a fix that's somewhere else entirely. As for the dot-notation, that's just how the DSR/ERAM/EDST/FDIO systems (used in the US) digest the routes. And it has the convenience factor of differentiating for you when it's linking two fixes, as opposed to taking the form of `entryPoint.procedureName.exitPoint`.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:311]: so this is actually `ONE_HOUR_IN_SECONDS = 3600`?

That makes sense.  Time is hard, especially in javascript where time collides with the browser and a current user's timezone settings.  Often times you will see time handled in milliseconds, which is what I had assumed.  I'm glad I asked because I guessed wrong.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js:24]: Hmm, thats different than what I was thinking.

I thought I'd seen some logic in here somewhere with a potential route written as `A.J123.B..E.J130.G`

Is that even a thing?

I guess what I am getting at with origin/destination is in terms of the route definition in arrivals section of an airport json (from klas). `BETHL.GRNPA1.KLAS`.  Perhaps better names would be entryFixName and exitFixName where the entry is joining a route and exit is leaving it.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:281]: hmm. I'm going to have to look at this a little more.  Thanks for the info here, it certainly helps clarify things a little bit!

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:311]: yup, that would work. almost all time units I've stumbled across are stored here in seconds, for example `prop.game.time` (or whatever it may be called now) and the various timeouts are all second-based.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [reply src/assets/scripts/airport/Route/RouteModel.js:24]: Yeah, that could be a thing. And yes, that is better verbage, particularly because `origin` and `destination` should eventually be used for the origin and destination airports. But yes, for a further example, the following is also acceptable:

```
Given that A-B-C-D-E-F-G-H are arranged sequentially in a circle shape, which is an airway called J123:

"A.J123.C..G..H.J123.E..F.J123.D..A..B..C..D.J123.H" is a valid route (albeit a squirrely one)

Disclaimer: You couldn't actually have a circular airway; it is just for example purposes
```

Any combination of things are fine; all you can't do is single-link with non-procedures (eg `A.B.C.D.E` or double-link procedures `A..J123.E`.

**Comment by n8rzz** [comment]
@erikquinn I'm at a good stopping point on this one.  All of the heavy work is done for this and #19.  I would like you to take a look at it before I merge this in because there is a lot of new stuff here.

There will probably be more to do with StandardRoutes and Fixes once I get to the FMS stuff, but that is still a ways off.

File: src/assets/scripts/airport/AirportModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/AirportModel.js:41]: Thought for the future (that I'd imagine you might've already had yourself):

Values like these might probably be best stored in a central configuration file, so various things can be changed globally in the system, and then multiple configuration files could be chosen from in order to implement different things. Most notably this would be with settings related to the canvas, that define the shape/size/color of data blocks, targets, position symbols, history dots, airspace boundary markings, shading, video map markings, SUA boundary markings, etc. But keeping with that theme, having some behavioral defaults such as default initial altitudes and such might fit in well with that concept.

File: src/assets/scripts/airport/AirportModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/AirportModel.js:142]: Would `airport.sids` and `airport.stars` not be deprecated by `airport.sidCollection` and `airport.starCollection`? Off the top of my head, I don't believe the _data_ itself parsed from the airport files is really of any use once it's built into the `StandardRouteCollection`, from which the various methods are called. I'm not sure, but just figured I'd bring it up since I see both members are present.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js:44]: I'm not sure I follow..... I wouldn't have thought this would work, to just say `FixCollection.findFixByName(fixName)`, because `FixCollection` is a class (I assume), and not an actual _instance_ of the class, so how could it access any fixes that would have been added to the collection? I've done things like `Object.keys(obj)` of course, but I wouldn't have thought this would do anything of use; I'm curious if you could briefly explain the intent here (if you have the time and patience to deal with me, that is!)

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js:54]: Not sure if it's been on the horizon or what, but the way all this was set up seemed strange to me, with these different classes, and each one inheriting `ArrivalBase` and then applying their own corrected methods. Since you are the simplification guy, I'm curious about your thoughts on how this could be arranged better, perhaps as separate methods of a singular class, or whatever the case may be. Again, little detail needed, just a concept I'm curious about.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js]: In some of the implementations, you could supply a range of altitudes at which to spawn aircraft, for example `[10000, 15000]`, or simply a singular altitude, `13000`. This block is just to keep the format of `this.altitude` consistent.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (erikquinn) [inline src/assets/scripts/airport/Arrival/ArrivalBase.js:201]: Agreed. In fact, the whole fix-array deal could theoretically just be removed. If users wanted their airports to just have arrival streams follow a string of fixes, there's no reason to not just make it a `route`, using dot-notation, and then have that formatted into a `RouteModel` just the same as if it had airways or STARs or any other procedures built into it.

File: src/assets/scripts/airport/Fix/FixCollection.js
Comment (erikquinn) [inline src/assets/scripts/airport/Fix/FixCollection.js:105]: Ooooooooooohh aaaaahhhhhhhhh
`instanceof` <-- I'm learning more every minute here lol

File: src/assets/scripts/airport/Fix/FixModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Fix/FixModel.js:27]: Sorry if it's becoming annoying, but I'm curious again here: what is the need to differentiate between whether or not a given instance of a `FixModel` is a particular instance as oppposed to another? I assume there must be situations where that is helpful, but I'm not sure what an example would be.

Comment (erikquinn) [reply src/assets/scripts/airport/Fix/FixModel.js:27]: As I read further, I guess the question is why `clonePosition` is needed. Why is that better to create a deep copy of the position object instance than a simple getter that retrieves and returns the position?

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Route/RouteModel.js]: At first, I went "oh now that will definitely not work sometimes". Then I thought through all the possible situations and found I was wrong...

Simple, and brilliant! :+1:

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Route/RouteModel.js:120]: Why the underscore prefix for some of these methods? I know that's a thing when calling lodash functions, but that's clearly not what this is. I've also noticed it with various classes' `._id` property. What's the story on the underscore?

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Route/RouteModel.js]: sep_A_ration

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Route/RouteModel.js:125]: Reference the `base` property, this might be made more clear to be renamed, perhaps `procedure`?

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/Route/RouteModel.js]: (presumably) has to do with the assumptions of the format of the route string. A single leg MUST have either `firstfix..secondfix` OR `firstfix.procedure.secondfix`. The former is handled somewhere else I guess, and the latter must have EXACTLY three elements when `.split(".")` is run on it. Otherwise, the route wouldn't make sense. So perhaps:
`const expected_elements_in_route_string = 3;`
or something more sensibly named (I'm getting sleepy and it's the best I can do)

File: src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js
Comment (erikquinn) [inline src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js:18]: This usage of `ICAO` doesn't really make sense. It's been (mis)used throughout the application in various places for a while actually. [ICAO](http://www.icao.int/Pages/default.aspx) is essentially the United Nations of airplane stuff, and they maintain their ICAO Document 8643, which contains unique and globally accepted identifiers for aircraft types, and Document 7910 for airport identifiers, etc. So it makes sense in that context, but it has in the app been kind of taken to be synonymous with `identifier` or `name`, which it is not. Not that it really matters, but just thought I'd share.

File: src/assets/scripts/airport/AirportModel.js
Comment (erikquinn) [reply src/assets/scripts/airport/AirportModel.js:41]: I see you have already done some of this, as with `airportConstants.js`.

File: src/assets/scripts/math/flightMath.js
Comment (erikquinn) [inline src/assets/scripts/math/flightMath.js:173]: This would probably make a lot more sense with the arguments reversed. I would tend to want to invoke this function as `calculateHeadingFromTwoPositions(aircraft.position, another.position);`, and would expect the heading FROM `aircraft` TO the other position, but here I would get the reverse. With `vsub()` it would make sense to feed parameters as "final minus initial", but with the more user-friendly named function, I definitely think the reverse is more intuitive.

**Review by erikquinn — APPROVED** [review]
I looked through everything, and it all makes sense. I left some questions and comments; do with them what you please, but overall I think this is a massive step forward, and nothing jumps out at me. I haven't actually transpiled and played the thing, but as far as the rearrangement, it still should work the same, whilst being hella better "orsganized".

**Comment by n8rzz** [comment]
@erikquinn I can't tell you how excited I was to see your comments and questions rolling in last night.  You asked some good questions for which I will be replying to as soon as possible.  Unfortunately, github was giving me 500s so I couldn't answer them last night.  Things look back to normal now.  I will be answering all of your questions today, but I might not get to all of them until this evening.  I will answer them all, though, so sit tight.

Keep asking questions!  I promise I will never make you feel stupid for asking a question, ever.  So don't feel shy about asking why!!

File: src/assets/scripts/airport/AirportModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/AirportModel.js:41]: You are correct, some (though not all) _do_ belong in a more global context.  I have an issue outstanding for that very thing #49.

The thing with constants is that you want them to be only as global as they need to be.  Maybe they are needed for a specific function, or a class or even the entire app.  It's sometimes hard to tell what the scope should until you start using them.

But yes, there will be more and they will be global.

File: src/assets/scripts/airport/AirportModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/AirportModel.js:142]: This is a good catch.  Yes, these two should not bee needed anymore.  However, before removing them completely I want to be sure they are not used elsewhere and that I have successfully moved everything to the new collections. Added a new issue for that #54

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:44]: To be honest, I still don't like this.  What I've done here is created a function, that can be used in the `ArrivalBase` class that then makes a call to the `FixCollection`.  I could have used ln 44 inline with exactly the same results.

But why does this work, you ask?  

The `FixCollection` is a very special class.  It is a _static_ class.  What that means is that there is only ever _one_ instance of this class.

Normally when we create a class it is defined thusly:

``` javascript
export default class Thing {
    doSomething() { }
    // ...
}
```

and then we use it:

``` javascript
import Thing from './Thing';

const thingInsance = new Thing();
thingInstance.doSomething();
```

but with a static class, things are slightly different:

``` javascript
class Thing {
    doSomething() { }
    // ...
}

export default new Thing();
```

and then we use it:

``` javascript
import Thing from './Thing';

Thing.doSomething();
```

These two are slightly different in use and drastically different in implementation.  By exporting `export default new Thing()` we create only one instance of the class.  That means different classes in the app can import the `FixCollection` and all will be able to work with the _exact same_  instance with the _same_ data.  This is important with the `FixCollection` because there is only ever one instance of that data set.  And several classes need access to it.

Currently, this is the only static class in the app, though I will be refactoring a few of the other classes to behave this way (CanvasController, GameController, etc).

This isn't a pattern I use often.  It has it's place, though, and this is one of them.

Does all that make sense?

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:54]: In this case, I think this actually makes sense.  The factory figures out which class to instantiate and then that class has only what it needs to create arrivals.  

We could move more of the shared properties to `ArrivalBase`?  There are several that are shared by the inheriting classes that could easily be moved. 

Created an issue for that #55

File: src/assets/scripts/airport/Fix/FixCollection.js
Comment (n8rzz) [reply src/assets/scripts/airport/Fix/FixCollection.js:105]: so Javascript is sort of the wild-west of programming languages.  You can do, pretty much, whatever you want and will just hum along.  

`instanceof` is a very handy little operator.  What I am doing here is a little bit of type checking.  

We have a class `FixModel` that under the hood calls a `constructor` function that is specific to this particular class.  If you pull open dev tools and inspect a `FixModel` object you will see that it has a name `FixModel` and a constructor function also named `FixModel`:
screen shot 2016-11-01 at 7 07 07 am

What `instanceof` does is basically compares types, or actually constructor functions.  In the example above, if the `fixToAdd` is not an instance of a `FixModel`, I don't want to add it to the `FixCollection`.  This is an example of defensive programming, I'm just making sure nothing gets added to this collection that shouldn't be.  Without this line, I could just as easily do this:

``` javascript
FixCollection.addFixToCollection(false);
FixCollection.addFixToCollection(42);
FixCollection.addFixToCollection('incorrect data');
```

Obviously, that isn't what we want in the collection, we want `FixModel`s and only `FixModel`s.  There is a test for this, actually:

``` javascript
ava.serial('.addFixToCollection() throws if it doesnt receive a FixModel instance', t => {
    t.throws(() => FixCollection.addFixToCollection({}));
});
```

for more info on `instanceof` see:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof

File: src/assets/scripts/airport/Fix/FixModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Fix/FixModel.js:27]: 1)  That is a good question; that isn't actually what is happening here.  We are checking that a thing passed in is a `FixModel` instance.  Not that is a specific instance in a group of instances.  See above for more on this one.  Let me know if have more questions on this because it's a little hard to grasp.

2) We could do that, in fact we _do_ do that.  The `FixModel` class has a getter for that very thing:

``` javascript
get position() {
    return this._fixPosition.position;
}

// which is used elsewhere as
const fix = new FixModel();
fix.position;
```

And we use that in a few places.  However, we are doing something slightly different in `clonePosition`.  Take a look at the `_init` method:

``` javascript
_init(fixName, fixCoordinate, airportPosition) {
    this.name = fixName.toUpperCase();
    this._fixPosition = new PositionModel(fixCoordinate, airportPosition, airportPosition.magneticNorthInRadians);

    return this;
}
```

the `_fixPosition` property is a `PositionModel` so we are creating a new instance of a `PositionModel` here.  Then in the `clonePosition` method:

``` javascript
/**
  * Returns a clone of an instance's `_fixPosition` property.
  *
  * It is important to note that this is a _clone_ and not a copy. Once any changes made to this instance will
  * not be reflected in the clone. This creates an entirely new instance of the `_fixPosition` property,
  * and after creation is completely independant of this instance.
  *
  * This is used with `StandardRouteWaypointModel` objects to obtain the position of a fix. This method
  * provides easy access to the `PositionModel` that already exists here.
  *
  * @for FixModel
  * @return {PositionModel}  a clone of the current `_fixPosition` property
  */
clonePosition() {
    return _cloneDeep(this._fixPosition);
}
```

What we are doing here is essentially: `return new PositionModel(this._fixPosition);`.  This creates an entirely new instance of a `PositionModel` from the data that already exists in `_fixPosition`.  This means that anything calling this method now has access to the methods that live in the `PositionModel` as well as the current information contained in the `_fixPosition` property.

Does that make sense?

I wrestled with how to do this for a while.  I wanted the `StandardRouteWaypointModel` to have actual position information but I didn't want it to be an extension of a `FixModel` because the two are fundamentally different, though they share some similar data.  This is a way around that.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js]: Thank you.

In its current implementation, yes, it will work all the time.  I know this because I have tests that prove it.  But...

Now that I have begun working through the fms, `Leg` and `Waypoint` models, I can see that this class will need some work.  Thankfully, the foundation is there.  So we can say goodbye to:

``` javascript
const entry = this.route.split('.')[0];
const star = this.route.split('.')[1];
const apt = this.route.split('.')[2];
```

It now reads better and is encapsulated in a central spot.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js:120]: Two things here I'll start with the second first:  why `_id`?  This is an old habit of mine, you will probably see this in models that will have many instances.  They are not being used at all but provide a way to differentiate instances.  Its helpful when inspecting to have an id for each model.  Who nows, eventually we may want to implement a findById method?  By having the ids in place already, that work would be trivial.

'Why the underscore?'
Remember how I said that Javascript is like the wild west?  This is another example.  the underscore here is a convention for indicating when something is private.  Ill show you some examples of what that means:

``` javascript
export default class RouteModel {
    canIPleaseDoThisThing() { 
        return this._youMayProceed();
    }

    _youMayProceed() { }
}

// which would be used as:
const route = new RouteModel();
route.canIPleaseDoThisThing();
```

The above is a common architectural pattern that I use.  Essentially I provide public methods that are really just wrappers for private methods.  The idea of a private method/variable, is that it is used only within that specific class by that specific class.  From the above example, this would be incorrect: `route._youMayProceed();`.  Now there isn't actually anything stopping you from doing that, per-se, but it is not the expected interface.  Same thing goes for private variables.  From the `FixModel` above:

``` javascript
get position() {
    return this._fixPosition.position;
}
```

Any external method should not access private methods or variables.  Why is that you ask?  Think about it like this:  I'll give you a way to get into the bank, but you have to speak with the teller (public method) in order for you to get money out of the bank.  How the teller gets that money is of no concern to you (private methods).

Feel free to ask more questions on this one, public/private methods can be confusing.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js]: :-(

Spelling is hard.  Seriously, spelling bites me in the but more often than not. grrrr.

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js:125]: Sure, I can do that.

File: src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js:18]: No, this does matter, and I agree icao does not make sense here.  Naming is one of the hardest parts of programming, thus things should be named accurately.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js]: excellent, I'll update this comment to call that out.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js:201]: Added #56 to capture this refactor.  

Wether this goes away completely or is just simplified has yet to be determined, but at least there is an issue for it.

File: src/assets/scripts/math/flightMath.js
Comment (n8rzz) [reply src/assets/scripts/math/flightMath.js:173]: You're right, however, I merely renamed a previous implementation so it would be inline with current usages.  So the order of properties came from that.  

I agree, though, that the order is confusing.  I'll add a comment here to clarify the why and hopefully that help make things a little clearer.

**Comment by n8rzz** [comment]
I plan to update this with the changes you called out, and those that don't have a new issue created for them.  So, spelling and comments, etc.  Everything else will be covered under the newly created issues.

File: src/assets/scripts/airport/Arrival/ArrivalBase.js
Comment (n8rzz) [reply src/assets/scripts/airport/Arrival/ArrivalBase.js]: done

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js]: done

File: src/assets/scripts/airport/Route/RouteModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/Route/RouteModel.js:125]: done

File: src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js
Comment (n8rzz) [reply src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js:18]: added #57 for this one.

**Pull request merged**
